src/Ast.ml

```diff
@@ -2215,6 +2213,8 @@ end = struct
     | Ptyp_object _ -> true
     | _ -> false
 end
+(** Operations determining precedence and necessary parenthesization of
+    terms based on their super-terms. *)
```

Comment:
module_expr_is_simple should not return true for this kind of module, we need to recursively call this function for Pmod_apply and Pmod_constraint (maybe other cases too if it makes sense, I didn't try it)

Reply:
I improved this. I added some arbitrary rules (eg. 2 or more `with`-constraints is not simple, extensions are not simple, etc..). What do you think ?